app/src/db/migrations/20220130133615_000-init.js

```diff
@@ -64,6 +64,97 @@ exports.up = function (knex) {
       stamps(knex, table);
     }))
 
+    // Audit tables for object and object permissions
+    .then(() => knex.schema.createTable('audit_object', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY'
+      );
+      table.uuid('objectId').notNullable().index();
+      table.string('dbUser').notNullable();
+      table.string('updatedByUsername');
+      table.timestamp('actionTimestamp', { useTz: true }).defaultTo(knex.fn.now());
+      table.string('action').notNullable();
+      table.jsonb('originalData');
+    }))
+
+    .then(() => knex.schema.createTable('audit_object_permission', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY'
+      );
+      table.uuid('objectPermissionId').notNullable().index();
+      table.string('dbUser').notNullable();
+      table.string('updatedByUsername');
+      table.timestamp('actionTimestamp', { useTz: true }).defaultTo(knex.fn.now());
+      table.string('action').notNullable();
+      table.jsonb('originalData');
+    }))
+
+    .then(() => knex.schema.raw(`CREATE OR REPLACE FUNCTION public.audit_object_func() RETURNS trigger AS $body$
+    DECLARE
+        v_old_data json;
+    BEGIN
+        if (TG_OP = 'UPDATE') then
+            v_old_data := row_to_json(OLD);
+            insert into public.audit_object ("objectId", "dbUser", "updatedByUsername", "actionTimestamp", "action", "originalData")
+            values (
+            OLD.id,
+            SESSION_USER,
+            NEW."updatedBy",
+            now(),
+            'U',
+            v_old_data);
+            RETURN NEW;
+        elsif (TG_OP = 'DELETE') then
+            v_old_data := row_to_json(OLD);
+            insert into public.form_submission_audit ("objectId", "dbUser", "actionTimestamp", "action", "originalData")
+            values (
+            OLD.id,
+            SESSION_USER,
+            now(),
+            'D',
+            v_old_data);
+        end if;
+    END;
+    $body$ LANGUAGE plpgsql`))
+
+    .then(() => knex.schema.raw(`CREATE OR REPLACE FUNCTION public.audit_object_permission_func() RETURNS trigger AS $body$
+    DECLARE
+        v_old_data json;
+    BEGIN
+        if (TG_OP = 'UPDATE') then
+            v_old_data := row_to_json(OLD);
+            insert into public.audit_object ("objectPermissionId", "dbUser", "updatedByUsername", "actionTimestamp", "action", "originalData")
+            values (
+            OLD.id,
+            SESSION_USER,
+            NEW."updatedBy",
+            now(),
+            'U',
+            v_old_data);
+            RETURN NEW;
+        elsif (TG_OP = 'DELETE') then
+            v_old_data := row_to_json(OLD);
+            insert into public.form_submission_audit ("objectPermissionId", "dbUser", "actionTimestamp", "action", "originalData")
```

Comment:
should be insert into table 'audit_object_permission'

Reply:
Oh nice catch. Thanks!